src/Main.elm

```diff
@@ -1,10 +1,396 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
+import Browser
+import Data.Audience exposing (Audience)
+import Data.AudienceFolder exposing (AudienceFolder)
+import Data.AudienceTree exposing (AudienceTree)
 import Html exposing (Html)
+import Html.Attributes
+import Html.Events
+import Json.Decode exposing (Decoder)
+import Process
+import Task
 
 
-main : Html msg
+type Model
+    = Loading LoadingModel
+    | DecodingError
+    | Loaded LoadedModel
+
+
+type Msg
+    = LoadingMsg LoadingMsg
+    | LoadedMsg LoadedMsg
+
+
+{-| simulates a http loading
+-}
+wait : Msg -> Cmd Msg
+wait msg =
+    Task.perform (always msg) (Process.sleep 1000)
+
+
+listInData : Decoder a -> Decoder (List a)
+listInData decoder =
+    Json.Decode.field "data" (Json.Decode.list decoder)
+
+
+init : () -> ( Model, Cmd Msg )
+init _ =
+    ( Loading { audiences = Nothing, audienceFolders = Nothing }
+    , Cmd.batch
+        [ GotAudiences
+            (Json.Decode.decodeString
+                (listInData Data.Audience.decoder)
+                Data.Audience.fixtures
+            )
+            |> LoadingMsg
+            |> wait
+        , GotAudienceFolders
+            (Json.Decode.decodeString
+                (listInData Data.AudienceFolder.decoder)
+                Data.AudienceFolder.fixtures
+            )
+            |> LoadingMsg
+            |> wait
+        ]
+    )
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case ( msg, model ) of
+        ( LoadingMsg loadingMsg, Loading loading ) ->
+            updatedLoading loadingMsg loading
+
+        -- instead of using the `_` wildcard for all the other possible
+        -- (msg, model) combinaison, I pattern match against LoadingMsg and
+        -- "anything else". This way, if I add some variant to the Msg
+        -- type, the compiler will help me finding out where to add code.
+        ( LoadingMsg _, _ ) ->
+            ( model, Cmd.none )
```

Comment:
Ahh, [where have I seen this](https://github.com/elm-in-elm/compiler/pull/68) :smile: 

Nice!

Reply:
I use this trick as much as I can now since I discovered it in elm-in-elm!